AnimatedComponents/src/components/floating-button/floating-button-with-label/FloatingButtonWithLabel.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import { Image, Pressable, View } from 'react-native'
+import Animated, {
+  Easing,
+  Extrapolation,
+  SharedValue,
+  interpolate,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withDelay,
+  withSpring,
+  withTiming,
+} from 'react-native-reanimated'
+
+import { PlusIcon } from '@assets'
+import { FloatingButtonProps } from '@types'
+
+import styles from './floatingButtonWithLabel-styles'
+
+const FloatingButtonWithLabel = (props: FloatingButtonProps) => {
+  const {
+    animationTransitionDuration = 500,
+    buttonContainerStyle = styles.buttonContainer,
+    contentContainerStyle = styles.contentContainer,
+    icons,
+    iconContainerStyle = styles.iconContainer,
+    iconStyle = styles.icon,
+    textStyle = styles.text,
+    isLeft = false,
+  } = props
+
+  const isOpen = useSharedValue(false)
+  const opacity = useSharedValue(0)
+  const progress = useDerivedValue(() => (isOpen.value ? withTiming(1) : withTiming(0)))
+  const length = icons?.length
+  const sharedValues: SharedValue<number>[] = []
+  const sharedWidthValues: SharedValue<number>[] = []
+
+  for (let i = 0; i < length; i += 1) {
+    sharedValues.push(useSharedValue(30))
+  }
+
+  for (let i = 0; i < length; i += 1) {
+    sharedWidthValues.push(useSharedValue(60))
+  }
+
+  const handlePress = () => {
+    const config = {
+      easing: Easing.bezier(0.68, -0.6, 0.32, 1.6),
```

Comment:
why these specific values and calculation or percentage value?

Reply:
This is done for smooth down the start and end of the interpolation.